src/Symfony/Component/Runtime/README.md

```diff
@@ -0,0 +1,114 @@
+Runtime Component
+=================
+
+Symfony Runtime enables decoupling apps from global state.
+
+Getting Started
+---------------
+
+```
+$ composer require symfony/runtime
+```
+
+RuntimeInterface
+----------------
+
+The core of this component is the `RuntimeInterface` which describes a high-order
+runtime logic.
+
+It is designed to be totally generic and able to run any application outside of
+the global state in 6 steps:
+
+ 1. your front-controller returns a closure that wraps your app
+ 2. `RuntimeInterface::resolve()` is given this closure and returns a new closure
+    that wraps the original one but can be called without any arguments (the goal
+    of this step is to resolve the arguments of the original closure, typically by
+    creating domain objects from PHP superglobals;)
+ 3. this closure is called and returns an object that represents the userland app
+    (e.g a Symfony kernel or response, a console application or command;)
+ 4. this userland app object is passed to `RuntimeInterface::start()`, which
+    returns another closure that knows how to "run" the object;
+ 5. that closure is called and returns the exit status code as int;
+ 6. the PHP engine is exited with this status code.
+
+This process is extremely flexible as it allows implementations of
+`RuntimeInterface` to hook into any critical steps.
+
+Note that closures are used instead of callables for performance. This is not a
```

Comment:
why closures are faster than callables?

Reply:
I suppose turning a `[$obj, 'method']` requires more runtime checks. But this is easily verifiable, give it a try if you want. That's a pro-tip: if you need to use a callable on the hot-path, turn it into a closure before.

Reply:
Is it really a hot path? Every callable in Runtime is called only once per hit, if I get it right.

Reply:
I removed that sentence from the readme since that's out of topic for a readme.